pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -230,7 +242,7 @@ func (r *InferenceServiceReconciler) updateStatus(desiredService *v1beta1api.Inf
 	if err := r.Get(context.TODO(), namespacedName, existingService); err != nil {
 		return err
 	}
-	wasReady := inferenceServiceReadiness(existingService.Status)
+	wasReady := isvcLatestRevisionReadiness(existingService.Status)
```

Comment:
We probably need to change back here as the ready condition also covers the routesReady which may not necessarily mean for the latest revision.